### PR TITLE
Mars

### PR DESCRIPTION
**Pull request by mcenerney1**
This is the initial version of mars to be installed in genutil.

### CODE REVIEW

**Comment by doutriaux1** [comment]
@mcenerney1 test suite fails

File: .idea/vcs.xml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (doutriaux1) [inline .idea/vcs.xml]: what is the file about? we should not put it in code.

File: Lib/mars_tools/src/MarsBasisFunction.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        add = True
+        for s in self.splines:
+            add = add and (ls.var != s.var)
+        if add:
+            self.splines = self.splines + [ls]
+        else:
+            self.error = 'spline has the same variable as one in the list'
+        return add
+
+if __name__ == '__main__':
```
Comment (doutriaux1) [inline Lib/mars_tools/src/MarsBasisFunction.py]: this looks like a test, should be rewritten as such under unittest and in `tests` directory

File: Lib/mars_tools/src/MarsResponseSurface.py [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+            parent = i
+            while parent != 0:
+                #print parent
+                curbf = BFs[parent]
+                ls = LinearSpline(curbf['var'], curbf['knot'], curbf['sign'])
+                fnc.addSpline(ls)
+                parent = curbf['parent']
+            if coef != 0.0:
+                self.addBasisFunction(fnc)
+if __name__ == '__main__':
```
Comment (doutriaux1) [inline Lib/mars_tools/src/MarsResponseSurface.py]: test?

File: .idea/vcs.xml (hunk 1 above)
Comment (mcenerney1) [inline .idea/vcs.xml]: I don't know. I find on web a procedure to remove it.

File: Lib/mars_tools/src/MarsBasisFunction.py (hunk 2 above)
Comment (mcenerney1) [inline Lib/mars_tools/src/MarsBasisFunction.py]: This was only intended to be used for testing changes to the class.

File: Lib/mars_tools/src/MarsResponseSurface.py (hunk 3 above)
Comment (mcenerney1) [inline Lib/mars_tools/src/MarsResponseSurface.py]: same answer

**Comment by mcenerney1** [comment]
Preserving metadata is not included but it would be useful.  The unit tests would need it. There was never any capability of referencing axes by name. This would be useful. I'll create unit tests for mars.

**Pull request closed without merging**